zktest/test/java/org/zkoss/zktest/zats/test2/F86_ZK_3958Test.java

```diff
@@ -0,0 +1,72 @@
+/* F86_ZK_3958Test.java
+
+        Purpose:
+                
+        Description:
+                
+        History:
+                Wed Jul 18 11:49:16 CST 2018, Created by charlesqiu
+
+Copyright (C) 2018 Potix Corporation. All Rights Reserved.
+*/
+package org.zkoss.zktest.zats.test2;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.python.parser.ast.Str;
+
+import org.zkoss.zktest.zats.WebDriverTestCase;
+import org.zkoss.zktest.zats.ztl.JQuery;
+
+public class F86_ZK_3958Test extends WebDriverTestCase {
+	
+	private JQuery isCheckedButton;
+	
+	@Test
+	public void test() {
+		connect();
+		isCheckedButton = jq(".z-button:contains(isChecked)");
+		JQuery moldLocation = jq(".z-checkbox-mold");
+		String switchClass = ".z-checkbox-switch";
+		String toggleClass = ".z-checkbox-toggle";
+		String offClass = ".z-checkbox-off";
+		String onClass = ".z-checkbox-on";
+		
+		verifyClassChange(jq(".z-button:contains(Switch)"), jq(switchClass + offClass));
+		Assert.assertFalse(jq(".z-checkbox > input").isVisible());
+		
+		verifyChecked("false");
+		
+		verifyClassChange(moldLocation, jq(switchClass + onClass));
+		
+		verifyChecked("true");
+		
+		verifyClassChange(jq(".z-button:contains(Toggle)"), jq(toggleClass + onClass));
+		
+		verifyClassChange(moldLocation, jq(toggleClass + offClass));
+		
+		verifyChecked("false");
+		
+		click(jq(".z-button:contains(setDisabled)"));
+		waitResponse();
+		Assert.assertEquals("0.5", moldLocation.css("opacity"));
```

Comment:
Not good for customizing.

Preferred: `setDisabled` toggleClass('z-checkbox-disabled'). You check if the class is applied.